src/problems.jl

```diff
@@ -90,20 +93,15 @@ function objective_vexity(p::Problem)
     return obj_vex
 end
 
-vexity(p::Problem) = objective_vexity(p)
-
-# is this right?
-# function problem_vexity(sense, obj_vexity, constr_vexity)
-#     if constr_vexity == ConstVexity()
-#         return obj_vexity
-#     end
-
-#     if constr_vexity == AffineVexity() && obj_vexity == AffineVexity()
-#         return sense === :maximize ? ConcaveVexity() : ConvexVexity()
-#     end
-
-#     return obj_vexity + constr_vexity
-# end
+function vexity(p::Problem)
```

Comment:
So `vexity(::Problem)` is really just "what is the verity of the objective function"

There is also `objective_vexity` which returns `Convex` if you are maximizing a concave function, and there is `problem_vexity` which also incorporates the constraints.

`solve!` uses `problem_vexity` to check for DCP-ness.

Reply:
thank you for sorting this out!